packages/auto-approve/src/auto-approve.ts

```diff
@@ -56,6 +56,56 @@ export async function authenticateWithSecret(
   return new Octokit({auth: payload});
 }
 
+/**
+ * Grabs an etag for a PR, and retries adding a label if etag is different
+ *
+ * @param retryCount the amount of retries for adding a label
+ * @param owner string, of the repo of the incoming PR
+ * @param repo string, the name of the repo of the incoming PR
+ * @param prNumber number, the number of the PR
+ * @param octokit the octokit instance
+ */
+export async function retryAddLabel(
+  retryCount: number,
```

Comment:
A more common pattern is to provide the number of retries remaining, rather than the retry attempt number. A private helper might keep the current attempt count as an argument. It's a bit weird for the caller to always provide the 0 argument on the initial call and have no control over the number of attempts.